sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/ErrorReceiver.java

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.common;
+
+/**
+ * An {@code ErrorReceiver} is a class that can be used to report errors on transfers. When specified on transfer
+ * operations, the {@code reportError} method will be called whenever errors are encountered. The user may configure
+ * this method to report errors in whatever format desired.
+ */
+public interface ErrorReceiver<T> {
```

Comment:
I think we could replace this with https://docs.oracle.com/javase/8/docs/api/java/util/function/Consumer.html in respective API and just call parameters `errorConsumer`.

Reply:
Ya in .NET we replaced this interface with a lamda.

Reply:
Do you think it's not important to follow the pattern of Progress Receiver?

Reply:
I think so. In next major rev we could think about using standard Java functional interfaces for Progress receiver as well. 
Let's not create types where standard JDK types can be used.

Reply:
@KrzysztofCwalina wasn't a fan of the interface in .NET:

https://github.com/Azure/azure-sdk-for-net/pull/11586#discussion_r423212648

Reply:
Ok I'm cool with using a consumer interface.

Reply:
I also changed the ProgressReceiver and made a custom BlobQueryProgress class